README.md

```diff
@@ -1,5 +1,347 @@
+production environment.
+
+The codebase takes advantage of [Vercel environment
+variables](https://vercel.com/docs/concepts/projects/environment-variables) at
+build time. For example, the preview mode banner links to the git branch that
+the site was built from (when that environment variable is available at build
+time).
+
+## Ways to view the site at different stages
+
+Throughout the develop-deploy process there are a number of ways to view the
+website. The following table summarizes the important ways in which those views
+differ from each other.
+
+| Name                       | How to access          | GitHub branch | Vercel env | Storyblok API key | Next.js preview? | Storyblok version param | Display top banner? | Top banner color | Button to enter/exit preview? |
+| -------------------------- | ---------------------- | ------------- | ---------- | ----------------- | ---------------- | ----------------------- | ------------------- | ---------------- | ----------------------------- |
+| Production                 | .com/.org URL          | `main`        | production | public            | off              | `published`             | No                  | n/a              | n/a                           |
+| Storyblok (yellow banner)  | via Storyblok UI       | `develop`     | preview    | preview           | on               | `draft`                 | Yes                 | gray             | No                            |
+| Vercel URL (gray banner)   | via link to Vercel URL | any non-main  | preview    | preview           | off              | `published`             | Yes                 | yellow           | Yes                           |
+| Vercel URL (yellow banner) | via enter-preview      | any non-main  | preview    | preview           | on               | `draft`                 | Yes                 | gray             | Yes                           |
+
+Let's take the row labeled "Vercel URL (gray banner)." This view is accessed by
+clicking on a Vercel SHA-style URL, which looks like
+https://quansight-labs-SHA-quansight.vercel.app. Typically this URL is found
+either on a GitHub pull request or in Slack. The site served by that URL can be
+built from any branch or commit on GitHub except `main`, which is reserved for
+production. It is built with the "preview" Vercel environment. It does not start
+in Next.js preview mode (though the end user can switch into it). The preview
+Vercel environment contains the preview Storyblok API key. The site passes
+`version=published` to the Storyblok API. It displays a gray banner at the top
+of each page. And the gray banner contains a way for the end user to switch in
+to and out of the Next.js preview mode.
+
+Note that within Storyblok, there is no button to switch out of preview mode.
+This was done by design because the whole point of the Storyblok UI is to be
+able to preview content that hasn't been published yet.
+
+## System design decisions
```

Comment:
Is this section mostly a collection of rationales for various decisions reflected in the sections above this point?

Asking for clarification, looking toward the future and the eventual organization/structure of all this content.

Reply:
yes that's correct